src/main/java/BreadthFirstSolve.java

```diff
@@ -1,59 +1,80 @@
-            boardsToCheck.add(moveL);
-        }
-
-        //move right
-        if (!currentBoard.getLastMove().equals("L") && currentBoard.getEmptyY() != 3
-                && currentBoard.getIterations() < ITERATIONS) {
-            FifteenPuzzle moveR = currentBoard.clone();
-            moveR.moveEmptyR();
-            boardsToCheck.add(moveR);
-        }
-
-        //move up
-        if (!currentBoard.getLastMove().equals("D") && currentBoard.getEmptyX() != 0
-                && currentBoard.getIterations() < ITERATIONS) {
-            FifteenPuzzle moveU = currentBoard.clone();
-            moveU.moveEmptyU();
-            boardsToCheck.add(moveU);
-        }
-
-        //move down
-        if (!currentBoard.getLastMove().equals("U") && currentBoard.getEmptyX() != 3
-                && currentBoard.getIterations() < ITERATIONS) {
-            FifteenPuzzle moveD = currentBoard.clone();
-            moveD.moveEmptyD();
-            boardsToCheck.add(moveD);
+        if (stats.checkAndDisplayStats(this.currentBoard)) return;
+        for (int i = 0; i < 4; i++) {
+            move(chars[i]);
         }
         this.currentBoard = boardsToCheck.poll();
         if (this.boardsToCheck.size() != 0) {
             this.solve();
         }
     }
+
+    private void move(char c) throws CloneNotSupportedException {
+        switch (c) {
+            case 'L':
+                //move left
+                if (!currentBoard.getLastMove().equals("R") && currentBoard.getEmptyY() != 0) {
+                    FifteenPuzzle moveL = currentBoard.clone();
+                    moveL.moveEmptyL();
+                    boardsToCheck.add(moveL);
+                    stats.incrementProcessed();
+                }
+                break;
+            case 'R':
+                //move right
+                if (!currentBoard.getLastMove().equals("L") && currentBoard.getEmptyY() != 3) {
+                    FifteenPuzzle moveR = currentBoard.clone();
+                    moveR.moveEmptyR();
+                    boardsToCheck.add(moveR);
+                    stats.incrementProcessed();
+                }
+                break;
+            case 'U':
+                //move up
+                if (!currentBoard.getLastMove().equals("D") && currentBoard.getEmptyX() != 0) {
+                    FifteenPuzzle moveU = currentBoard.clone();
+                    moveU.moveEmptyU();
+                    boardsToCheck.add(moveU);
+                    stats.incrementProcessed();
+                }
+                break;
+            case 'D':
+                //move down
+                if (!currentBoard.getLastMove().equals("U") && currentBoard.getEmptyX() != 3) {
+                    FifteenPuzzle moveD = currentBoard.clone();
+                    moveD.moveEmptyD();
+                    boardsToCheck.add(moveD);
+                    stats.incrementProcessed();
+                }
+                break;
+        }
+    }
+
+    public Stats getStats() {
```

Comment:
This method is not used, because of not splited frontend and backed part of Stats class.

This getter should return Stats object and from it frontend class should gets data, and display.